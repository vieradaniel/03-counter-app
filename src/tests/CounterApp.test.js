import { shallow } from "enzyme"
import CounterApp from "../CounterApp"
import React from 'react';

import'@testing-library/jest-dom';

let wrapper= shallow(<CounterApp  />);

beforeEach(()=>{

    wrapper = shallow(<CounterApp  />);

})

describe('<CounterApp /> ',()=>{

    test('CounterApp se debe mostrar correctamente',()=>{

        

        expect(wrapper).toMatchSnapshot();

    });
    test('CounterApp debe mostrar el valor por defecto de 100',()=>{

        
        
        const wrapper = shallow(
            <CounterApp 
                value = { 100 } 
            />
        );

        const valueText = wrapper.find('h2').text();
        expect(valueText).toBe('100');

    })

    test('debe de incrementar con el boton +1',()=>{

        const btn1 = wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('11');
    })

    test('debe de decrementar con el boton -1',()=>{

        const btn1 = wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('9');
    })

    test('debe de colocar el valor por defecto con btn reset',()=>{

        const wrapper = shallow(
            <CounterApp 
                value = { 105 } 
            />
        );

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');

        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        
        
        expect(counterText).toBe('105');
    })


})