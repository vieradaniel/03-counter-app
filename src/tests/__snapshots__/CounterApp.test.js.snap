// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp />  CounterApp se debe mostrar correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
